modules/Movecraft/src/main/java/net/countercraft/movecraft/craft/BaseCraft.java

```diff
@@ -91,7 +93,60 @@ public BaseCraft(@NotNull CraftType type, @NotNull World world) {
         materials = new Counter<>();
         audience = Audience.empty();
     }
+    //Start of Adding Craft Datatag System
+    public void setDataTag(String key, Object data) {
+      this.dataTags.put(key, data);
+    }
+    public void removeDataTag(String key) {
+      this.dataTags.remove(key);
+    }
+    public Object getDataTag(String key) {
+      return this.dataTags.getOrDefault(key, "null");
+    }
+    public Collection<Object> getAllTagValues() {
+      return this.dataTags.values();
+    }
+    public boolean hasDataKey(String key) {
+      return this.dataTags.containsKey(key);
+    }
+    public boolean hasDataValue(Object value) {
+      return this.dataTags.containsValue(value);
+    }    
+    //End of Added Craft Datatag System
```

Comment:
I have a few problems with this:
1. `Tags` is too easily confused with `net.countercraft.movecraft.util.Tags`.
2. A system like this should be put in `Craft` instead since this is merely an implementation of `Craft`, and the user should never need to have to know what `BaseCraft` is unless they are developing their own `Craft` subclasses.
3. Strings are a terrible way to do this, the HashMap performance of strings is lacking, and we are instead aiming to use more `NamespacedKey`s (see [CraftType](https://github.com/APDevTeam/Movecraft/blob/main/modules/api/src/main/java/net/countercraft/movecraft/craft/type/CraftType.java) for an example of this).  Namespaced keys have their hash function computed at compile time, meaning that they effectively just as fast as integer keys at runtime.
4. There's no need for `this.` on everything unless there is a conflict.